app/controllers/application_controller.rb

```diff
@@ -1,4 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   include SessionsHelper
+
+  private
+
+  def logged_in_user
+    return if logged_in?
+    store_location
```

Comment:
Giải thích dòng này

Reply:
Kiểm tra nếu người dùng đã log in thì trả về validates là thành công và thực hiện tiếp action hoặc validattion khác, nếu không thì lưu lại url của request get để foward người dùng đến trang đó sau khi đăng nhập thành công